Add FELightingSoftwareApplierInlines.h

donny-dont opened the pull request:
#### 33a29269d60facb4b3c80bd33d31c643b8a887e5
<pre>
Add FELightingSoftwareApplierInlines.h
<a href="https://bugs.webkit.org/show_bug.cgi?id=267602">https://bugs.webkit.org/show_bug.cgi?id=267602</a>

Reviewed by Said Abou-Hallawa.

`FELightingSoftwareApplier::LightingData` struct declares a number of
`inline` methods which were defined in `FELightingSoftwareApplier.cpp`.
However usage outside of the file happened in 272873@main. Unified
builds continued to function because the `.cpp` files using it ended up
in the same compilation unit, but non-unified builds failed with a
linker error.

Move the inlined `FELightingSoftwareApplier::LightingData` methods to
a new file `FELightingSoftwareApplierInlines.h` and include them when
applicable.

* Source/WebCore/platform/graphics/filters/software/FELightingSoftwareApplier.cpp:
* Source/WebCore/platform/graphics/filters/software/FELightingSoftwareApplierInlines.h: Copied from Source\WebCore\platform\graphics\filters\software\FELightingSoftwareApplier.cpp.
* Source/WebCore/platform/graphics/filters/software/FELightingSoftwareParallelApplier.cpp:

Canonical link: <a href="https://commits.webkit.org/273112@main">https://commits.webkit.org/273112@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/998307518bd21beb1ad995abed806b6d2ada90a5

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/34316 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/13122 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/36308 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/36994 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/31059 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/35396 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/15500 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/51/builds/10238 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/30069 "Passed tests") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/34828 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/11107 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/30601 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/9691 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/42/builds/9789 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/10/builds/30647 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/38280 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/31170 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/30968 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/35864 "Passed tests") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/44/builds/9914 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/50/builds/7787 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/33792 "Passed tests") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/46/builds/11710 "Built successfully") | | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/7893 "Built successfully and passed tests") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/43/builds/10471 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/45/builds/10751 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

shallawa commented on Source/WebCore/platform/graphics/filters/software/FELightingSoftwareApplier.cpp:
```diff
@@ -29,156 +29,11 @@
+#include "FELightingSoftwareApplierInlines.h"
 #include "Filter.h"
-#include "ImageBuffer.h"
-#include "PixelBuffer.h"
```
I think methods of PixelBuffer are used in FELightingSoftwareApplier::applyPlatform(). So I think "PixelBuffer.h" is needed.

donny-dont replied:
When doing the patch I had a local non-unified build and I took out those to see if they were needed. It wouldn't surprise me if those were included somewhere else. Unfortunately we don't have any tooling to tell us whether the includes should be present or not 😞

donny-dont committed:
Add FELightingSoftwareApplierInlines.h
https://bugs.webkit.org/show_bug.cgi?id=267602

Reviewed by Said Abou-Hallawa.

`FELightingSoftwareApplier::LightingData` struct declares a number of
`inline` methods which were defined in `FELightingSoftwareApplier.cpp`.
However usage outside of the file happened in 272873@main. Unified
builds continued to function because the `.cpp` files using it ended up
in the same compilation unit, but non-unified builds failed with a
linker error.

Move the inlined `FELightingSoftwareApplier::LightingData` methods to
a new file `FELightingSoftwareApplierInlines.h` and include them when
applicable.

* Source/WebCore/platform/graphics/filters/software/FELightingSoftwareApplier.cpp:
* Source/WebCore/platform/graphics/filters/software/FELightingSoftwareApplierInlines.h: Copied from Source\WebCore\platform\graphics\filters\software\FELightingSoftwareApplier.cpp.
* Source/WebCore/platform/graphics/filters/software/FELightingSoftwareParallelApplier.cpp:

Canonical link: https://commits.webkit.org/273112@main
